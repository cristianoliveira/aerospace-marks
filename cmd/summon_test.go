package cmd

import (
	"strings"
	"testing"

	"github.com/cristianoliveira/aerospace-marks/internal/logger"
	"github.com/cristianoliveira/aerospace-marks/internal/mocks"
	"github.com/cristianoliveira/aerospace-marks/internal/testutils"
	"github.com/gkampitakis/go-snaps/snaps"
	"go.uber.org/mock/gomock"
)

func TestSummonCmd(t *testing.T) {
	t.Run("validate missing identifier", func(t *testing.T) {
		command := "summon"
		args := []string{command}

		ctrl := gomock.NewController(t)
		defer ctrl.Finish()

		logger.SetDefaultLogger(&logger.EmptyLogger{})

		_, strg := mocks.MockStorageDbClient(ctrl)
		_, aerospaceClient := mocks.MockAerospaceConnection(ctrl)

		cmd := NewRootCmd(strg, aerospaceClient)
		out, err := testutils.CmdExecute(cmd, args...)
		if out != "" {
			t.Fatal("output should be empty", out)
		}
		if err == nil {
			t.Fatal(err)
		}

		cmdAsString := "aerospace-marks " + strings.Join(args, " ") + "\n"
		snaps.MatchSnapshot(t, cmdAsString, "Error", err.Error())
	})

	t.Run("validate empty identifier", func(t *testing.T) {
		command := "summon"
		args := []string{command, " "}

		ctrl := gomock.NewController(t)
		defer ctrl.Finish()

		logger.SetDefaultLogger(&logger.EmptyLogger{})

		_, strg := mocks.MockStorageDbClient(ctrl)
		_, aerospaceClient := mocks.MockAerospaceConnection(ctrl)

		cmd := NewRootCmd(strg, aerospaceClient)
		out, err := testutils.CmdExecute(cmd, args...)
		if out != "" {
			t.Fatal("output should be empty", out)
		}
		if err == nil {
			t.Fatal(err)
		}

		cmdAsString := "aerospace-marks " + strings.Join(args, " ") + "\n"
		snaps.MatchSnapshot(t, cmdAsString, "Error", err.Error())
	})
}
