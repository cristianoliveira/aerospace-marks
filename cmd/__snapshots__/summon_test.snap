
[TestSummonCmd/validate_missing_identifier - 1]
aerospace-marks summon

Error
accepts 1 arg(s), received 0
---

[TestSummonCmd/validate_empty_identifier - 1]
aerospace-marks summon  

Error
argument cannot be empty or whitespace
---
